packages/material-ui/src/Select/SelectInput.js

```diff
@@ -141,6 +150,22 @@ const SelectInput = React.forwardRef(function SelectInput(props, ref) {
     event.preventDefault();
     displayRef.current.focus();
 
+    const doc = ownerDocument(event.target);
+    if (handleClick.current) {
+      doc.removeEventListener('click', handleClick.current, true);
+    }
+
+    // Since the targets of the `mousedown` and `mouseup` events are different,
+    // the browser will dispatch the `click` to a common ancestor.
+    // This stops the next `click` and re-dispatch it with the right target.
```

Comment:
Why do you think the specified behavior is not the "right" target?

Reply:
Because it's not clear that, because a transparent div is placed in the `mousedown`, the target of `mouseup` changes.